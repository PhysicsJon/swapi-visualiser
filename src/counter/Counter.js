import React, { Component } from 'react';

export default class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  onIncrement = () => {
    this.setState({ count: this.state.count + 1 });
  };

  onDecrement = () => {
    this.setState({ count: this.state.count - 1 });
  };

  render() {
    return (
      <div>
        <h2>Counter</h2>
        <h3 id="count">{this.state.count}</h3>
        <button onClick={this.onIncrement} data-test-id="add-counter">+</button>
        <button onClick={this.onDecrement} data-test-id="minus-counter">-</button>
      </div>
    );
  }
}
