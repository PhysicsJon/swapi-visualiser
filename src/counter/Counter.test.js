import { render, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Counter from './Counter';

describe('Counter rendering', () => {
  it('renders the counter correctly', () => {
    const component = renderer.create(<Counter />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Counter actions', () => {
  it('clicking `+` adds to count', () => {
    const { container } = render(<Counter />);
    const button = container.querySelector('[data-test-id=add-counter]');
    expect(button).not.toBeNull();

    act(() => {
      userEvent.click(button);
    });

    const count = document.getElementById('count');
    expect(count.innerHTML).not.toBeUndefined();
    expect(count.innerHTML).toBe('1');
  });

  it('clicking `-` minuses from count', () => {
    const { container } = render(<Counter />);
    const button = container.querySelector('[data-test-id=minus-counter]');
    expect(button).not.toBeNull();

    act(() => {
      userEvent.click(button);
    });

    const count = document.getElementById('count');
    expect(count.innerHTML).not.toBeUndefined();
    expect(count.innerHTML).toBe('-1');
  });
});
