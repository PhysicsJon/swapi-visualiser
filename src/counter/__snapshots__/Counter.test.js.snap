// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Counter rendering renders the counter correctly 1'] = `
<div>
  <h2>
    Counter
  </h2>
  <h3
    id="count"
  >
    0
  </h3>
  <button
    data-test-id="add-counter"
    onClick={[Function]}
  >
    +
  </button>
  <button
    data-test-id="minus-counter"
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
